freetextresponse/freetextresponse.py

```diff
@@ -393,18 +393,13 @@ def _compute_score(self):
         based on their answer
         """
         credit = self._determine_credit()
-        if credit == Credit.full:
-            self.score = self.weight
-        elif credit == Credit.half:
-            self.score = float(self.weight)/2
-        else:
-            self.score = 0.0
+        self.score = credit
         self.runtime.publish(
```

Comment:
I think a manifestation of this problem is that "score" in coursware_studentmodule.state has been getting set to weight e.g. '{"student_answer": "What do you expect of me? I have no instructions.", "score": 0, "count_attempts": 1}'
whereas I believe it should be set to the score. This is probably how it is able to now recalculate grade after the weight has been changed. Can you confirm my understanding is correct?